src/TypeScriptGenerator.ts

```diff
@@ -490,6 +524,101 @@ function createVisitor(options: TypeGeneratorOptions) {
   };
 }
 
+function visitCondition(node: Condition) {
+  return flattenArray(
+    /* $FlowFixMe: selections have already been transformed */
+    (node.selections as any) as ReadonlyArray<ReadonlyArray<Selection>>
+  ).map(selection => {
+    return {
+      ...selection,
+      conditional: true
+    };
+  });
+}
+
+function visitScalarField(node: ScalarField, state: State) {
+  return [
+    {
+      key: node.alias || node.name,
+      schemaName: node.name,
+      value: transformScalarType(node.type, state)
+    }
+  ];
+}
+
+function visitLinkedField(node: LinkedField) {
+  return [
+    {
+      key: node.alias || node.name,
+      schemaName: node.name,
+      nodeType: node.type,
+      nodeSelections: selectionsToMap(
+        flattenArray(
+          /* $FlowFixMe: selections have already been transformed */
+          (node.selections as any) as ReadonlyArray<ReadonlyArray<Selection>>
+        ),
+        /*
+         * append concreteType to key so overlapping fields with different
+         * concreteTypes don't get overwritten by each other
+         */
+        true
+      )
+    }
+  ];
+}
+
+function visitConnectionField(node: ConnectionField) {
+  return [
+    {
+      key: node.alias,
+      schemaName: node.name,
+      nodeType: node.type,
+      nodeSelections: selectionsToMap(
+        flattenArray(
+          /* $FlowFixMe: selections have already been transformed */
+          (node.selections as any) as ReadonlyArray<ReadonlyArray<Selection>>
+        ),
+        /*
+         * append concreteType to key so overlapping fields with different
+         * concreteTypes don't get overwritten by each other
+         */
+        true
+      )
+    }
+  ];
+}
+
+function makeRawResponseProp(
+  { key, schemaName, value, conditional, nodeType, nodeSelections }: Selection,
+  state: State,
+  concreteType: string | null
+) {
+  if (nodeType) {
+    if (nodeType === MODULE_IMPORT_FIELD) {
+      // TODO: In flow one can extend an object type with spread, with TS we need an intersection (&)
+      // return ts.createSpread(ts.createIdentifier(key));
+      throw new Error("TODO!");
```

Comment:
@kastermester FYI, here’s that `@match` todo

Reply:
Great - I might take a stab at this at some point. Now that I read that I do recall this from last time I ported the code.